src/som/interpreter/actors/RegisterOnPromiseNode.java

```diff
@@ -25,14 +37,57 @@ public void register(final SPromise promise, final PromiseMessage msg,
       // otherwise, we are opening a race between completion and processing
       // of handlers. We can split the case for a resolved promise out, because
       // we need to schedule the callback/msg directly anyway
+
       synchronized (promise) {
+        if (VmSettings.REPLAY) {
+          ReplayRecord npr = current.peekNextReplayEvent();
+          if (npr.type == TraceRecord.PROMISE_CHAINED && !promise.isResolvedUnsync()) {
+            npr = current.getNextReplayEvent();
+
+            assert msg instanceof AbstractPromiseSendMessage;
+            LinkedList<ReplayRecord> events = new LinkedList<>();
```

Comment:
Any specific reason why this is a `LinkedList`?

Reply:
Looked it up, anything that maintains order is fine.